### PR TITLE
[Delivers #91211506 #93936068] remove Cart dependency from client models

### PR DESCRIPTION
**Pull request by afeld**
I started this branch a week or so ago but put it aside for a while...this PR removes the Cart dependence from the WorkOrder and Procurement models. There's some other light refactoring, mostly taking advantage of the `ProposalDelegate` mixin and some better variable names. Sorry it's so monstrous :grin:

### CODE REVIEW

File: spec/factories/gsa18f/procurements.rb [hunk 1]
```diff
@@ -6,13 +6,7 @@
...
     office Gsa18f::Procurement::OFFICES[0]
     urgency Gsa18f::Procurement::URGENCY[0]
-    trait :with_proposal do
-      proposal
-    end
-
-    trait :with_cart do
-      association :proposal, :with_cart
-    end
+    proposal
```
Comment (cmc333333) [inline spec/factories/gsa18f/procurements.rb]: flow should be 'linear', hey?

**Comment by cmc333333** [comment]
Glad to see `Cart` be removed from more places.

Something to consider, with all of the nil wrappers for `client_data` in `Proposal`, you've created a pseudo base class. Part of my argument in #346 is that it's simpler to use language-supported inheritance.

File: spec/factories/gsa18f/procurements.rb (hunk 1 above)
Comment (afeld) [inline spec/factories/gsa18f/procurements.rb]: Good catch!

**Comment by afeld** [comment]
Updated!

> you've created a pseudo base class...it's simpler to use language-supported inheritance

That's fair. I still anticipate [other tradeoffs](https://github.com/18F/C2/pull/346#issuecomment-101361658), but this PR hopefully narrows the gap so that code outside of the model doesn't have to care as much about how the data is persisted.

**Pull request merged**